### PR TITLE
[modbus] Clarify poller names must be unique

### PR DESCRIPTION
**Pull request by Rossko57**
Clarify poller names must be unique  (seen several people fall into this trap)

### CODE REVIEW

**Comment by ssalonen** [comment]
Further testing here: https://community.openhab.org/t/2-or-more-modbus-slaves/46928/15?u=ssalonen

I'm personally not seeing any problems when having identically named poller ID?

**Comment by Rossko57** [comment]
Hmm, yes, this may be turning into a more general Thing editing issue.   Duplicate names may (accidentally) invoke "editing" processes, hence a potential confusion.

**Comment by Rossko57** [comment]
Had a play in **PaperUI** (I usually configure in xxx.things files)
Created two differing serial Bridge things. Different **thing ID** - serial4, serial 9
Next create a Regular Poll thing, bridged to serial4, with a thing ID of "samename".
(equivalent to text config `Bridge poller samename [ start=4,` ... )

Now attempt to create another Regular Poll, bridged to different serial9, but also with a thing ID of "samename".
_Looks_ feasible as the eventual channel IDs would be  modbus:data:serial**4**:samename:xxx , different to modbus:data:serial**9**:samename:xxx , right?
_And_ the "owner" bridge modbus:serial:slave4 is different to modbus:serial:slave9 as well?

Nope, PaperUI rejects this attempt with a popup message **ERROR 409 - conflict**

So, we are not permitted to create pollers with the same 'name' (meaning thing ID really) in PaperUI.
Looking at how PaperUI displays in the Configuration>Things listing, it is in a different form:
modbus:poller:samename
In that form, my attempt would be indistinguishable.  I suspect this is what PaperUI is complaining about duplicating.

Maybe PaperUI is enforcing something that it doesn't need to here?
Or maybe it's correct, and as a general principle all Bridge things need a unique ID?
I don't know.

But we certainly have something that you seem to able to do in xxx.things files (but perhaps goes wrong later), that you cannot do in PaperUI.

Thinking on @ Sami's experiment, I wonder if adding duplicate IDs from xxx.things file results in the creation/scheduling of new pollers one at a time (regardless of duplicate ID) which all looks fine until you edit one.

**Comment by ssalonen** [comment]
Ok, sounds sensible to avoid using duplicate poller bridges then. How about data things?

**Comment by Rossko57** [comment]
I'm guessing that some part(s) of OH has a requirement for **all** Bridge things, or all Bridges of one type, belonging to one binding to have unique IDs.  The fact that our Modbus Bridges are arranged in a hierarchy (where similar names can still have a unique 'pathway') is not taken into account.

Might not even be a design requirement, just some author's assumptions.  But we got what we got.

Interesting about data Things, will have to play later.

**Comment by Rossko57** [comment]
Played with PaperUI and data things, attempting the equivalent of
```
Bridge modbus:tcp:testmb [ host="192.168.1.12", port=502, id=24, connectMaxTries=2 ] {
    Bridge poller poller20 [ start=20, length=2, refresh=50000, type="holding" ] {
        Thing data samename [ readStart="20", readValueType="int16" ]
        }
    Bridge poller poller40 [ start=40, length=2, refresh=50000, type="holding" ] {
        Thing data samename [ readStart="40", readValueType="int16" ]
``
that's to say, duplicate data thing ID via different path / owning Bridge.
PaperUI rejects this attempt with a popup message **ERROR 409 - conflict**

For completeness, I tried duplicate data thing ID with different slaves _and_ pollers - same 409 rejection.

Interestingly, PaperUI _will_ let me make a data Thing with the same thing ID as a poller Bridge.  So it seems the ID must be unique only among similar types of thing.

Okay, so I don't know if this is PaperUI doing its own thing or if there is a fundamental reason for things of different parentage to have unique thing IDs  (though it seems likely).

**Comment by Rossko57** [comment]
I have made the "unique ID" requirement refer to thing ID more formally (as "name" is misleading), added also to data Thing description.  And I think signed off properly?!

**Comment by cweitkamp** [comment]
@ssalonen May I ask you for a final comment? Thanks.

File: bundles/org.openhab.binding.modbus/README.md [hunk 1]
```diff
@@ -1235,4 +1237,4 @@ Consult [openHAB2 logging documentation](http://docs.openhab.org/administration/
 
 ## For Developers
 
-This binding can be extended in many ways. If you have a Modbus enabled device that you want to support in openHAB please read the [developer section](DEVELOPERS.md) 
\ No newline at end of file
+This binding can be extended in many ways. If you have a Modbus enabled device that you want to support in openHAB please read the [developer section](DEVELOPERS.md)
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.modbus/README.md]: Please put each sentence into a separate line and add a full stop at the end.

```suggestion
This binding can be extended in many ways.
If you have a Modbus enabled device that you want to support in openHAB please read the [developer section](DEVELOPERS.md).
```

**Comment by ssalonen** [comment]
@cweitkamp thanks for the ping. LGTM.

Thanks @Rossko57 !

**Pull request merged**